spyder/utils/programs.py

```diff
@@ -259,16 +259,19 @@ def run_python_script_in_terminal(fname, wdir, args, interact,
 
     :str wdir: working directory, may be empty.
     """
-    # If fname has spaces on it it can't be ran on Windows, so we have to
-    # enclose it in quotes. Also wdir can come with / as os.sep, so we
-    # need to take care of it
+    python_exe = get_python_executable()
+
+    # If fname or python_exe contains spaces, it can't be ran on Windows, so we
+    # have to enclose them in quotes. Also wdir can come with / as os.sep, so
+    # we need to take care of it.
     if os.name == 'nt':
         fname = '"' + fname + '"'
         wdir = wdir.replace('/', '\\')
-    
-    p_args = [get_python_executable()]
+        python_exe = '"' + python_exe + '"'

```

Comment:
Maybe enclose in triple quotes to not run into #6451?  I'm not sure if it's necessary.

Reply:
Good idea @csabella.

Reply:
Good point. I'll take that into consideration and push a commit with the changes.